src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -185,63 +217,44 @@ private function handleDbError(\Exception $e, array $table = null, int $counter
         if ($counter) {
             $message .= sprintf(' Tried %d times.', $counter);
         }
-        $exception = new UserException($message, 0, $e);
-        return $exception;
+        return new UserException($message, 0, $e);
     }
 
-    /**
-     * @param $query
-     * @return int Number of rows returned by query
-     * @throws \PDOException|\ErrorException
-     */
-    protected function executeQuery($query, int $maxTries)
+    protected function executeQuery(string $query, ?int $maxTries): \PDOStatement
     {
-        $retryPolicy = new SimpleRetryPolicy($maxTries, ['PDOException', 'ErrorException']);
-        $backOffPolicy = new ExponentialBackOffPolicy(1000);
-        $proxy = new RetryProxy($retryPolicy, $backOffPolicy);
-        $counter = 0;
-        /** @var \Exception $lastException */
-        $lastException = null;
-        try {
-            $stmt = $proxy->call(function () use ($query, &$counter, &$lastException) {
-                if ($counter > 0) {
-                    $this->logger->info(sprintf('%s. Retrying... [%dx]', $lastException->getMessage(), $counter));
-                    try {
-                        $this->db = $this->createConnection($this->dbParameters);
-                    } catch (\Exception $e) {
-                    };
-                }
+        $proxy = new RetryProxy($this->logger, $maxTries);
+        $stmt = $proxy->call(function () use ($query) {
+            try {
+                /** @var \PDOStatement $stmt */
+                $stmt = @$this->db->prepare($query);
```

Comment:
what is @ good for here? doesn't it cripple the error handler?

Reply:
tbh I don't know, it's a relic.  even without the `@`, I am not able to get the prepare statement to throw any errors (they all throw on the execute statement) so I'll remove it.

Reply:
and in the `execute` it is necessary?

Reply:
I don't know, it's always been there.  I'd like to hear what @MiroCillik thinks about it.  My guess is that it's no longer needed, but it'd be good to know it's original purpose.

Reply:
it usually fucks up the error handler (though that one is modified too, so it might not be sensitive to it any more) - see here https://keboola.slack.com/archives/C035LGNK5/p1485473670000007

Reply:
you can throw an exception in prepare by this:
```
$pdo = new PDO('mysql:server=localhost;dbname=syrup', 'syrup', 'syrup');
$pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
$pdo->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
$pdo->prepare('gre23r23????');
```

to put that in tests, you'd have to fuck up with $this->db via reflection though, not sure if it is worth it, because on mysql  ATTR_EMULATE_PREPARES defaults to true